test/selector.spec.ts

```diff
@@ -7,6 +7,7 @@ import { select, validate, Path } from '../src/selector';
 const testData = {
```

Comment:
Most of this is just taking existing tests and adding in tests for quoting them.  Also empty string tests added.